lib/autoload/coursemology.rb

```diff
@@ -0,0 +1,7 @@
+module Coursemology
+  class ComponentNotFoundError < StandardError
```

Comment:
I also saw some other rails projects doing this, like: https://github.com/discourse/discourse/blob/master/lib/discourse.rb#L31

How about put them under exceptions ? (http://stackoverflow.com/questions/5200842/where-to-define-custom-error-types-in-ruby-and-or-rails)